src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -236,156 +234,142 @@ public unsafe bool ParseHeaders(TRequestHandler handler, ref SequenceReader<byte
                         length = span.IndexOf(ByteLF) + 1;
                         if (length > 0)
                         {
-                            // Potentially found the end, or an invalid header.
-                            fixed (byte* pHeader = span)
+                            if (length < 5 ||
+                                span[(length - 2)] != ByteCR ||
```

Comment:
```suggestion
                                span[^2] != ByteCR ||
```
to be consistent with other uses in this class?

Reply:
Wouldn't work, `length` is not `span.Length`

Reply:
Ahhh, sorry.